client/rest/test/plugins/metadata/metadataRoutes_spec.js

```diff
@@ -240,4 +240,45 @@ describe('metadata routes', () => {
 			});
 		});
 	});
+	describe('metal', () => {
+		const mockServer = new MockServer();
+		const db = {
+			binDataByMetalId: sinon.stub().resolves({ payload: 'db_payload', text: 'db_text' })
+		};
+		metadataRoutes.register(mockServer.server, db, {});
+
+		beforeEach(() => {
+			mockServer.resetStats();
+			mockServer.res.setHeader = sinon.spy();
+			mockServer.res.write = sinon.spy();
+			mockServer.res.end = sinon.spy();
+		});
+
+		describe('get by metal id', () => {
+			it('returns page with results', () => {
+				// Arrange:
+				const route = mockServer.getRoute('/metadata/metal/:metalId').get();
+				const metalId = 'your_metal_id';
+				const mimeType = 'image/png';
+				const fileName = 'image.png';
+				const download = 'true';
+				const req = {
+					params: {
+						metalId, mimeType, fileName, download
+					}
+				};
+				// Act:
+				return mockServer.callRoute(route, req).then(() => {
+					// Assert:
+					expect(mockServer.res.write.calledOnce).to.equal(true);
+					expect(mockServer.next.calledOnce).to.equal(true);
+					expect(mockServer.res.setHeader.calledWithExactly('content-type', 'image/png')).to.equal(true);
+					expect(mockServer.res.setHeader
+						.calledWithExactly('Content-Disposition', 'attachment; filename="image.png"')).to.equal(true);
+					expect(db.binDataByMetalId.calledOnce).to.equal(true);
+					expect(db.binDataByMetalId.alwaysCalledWith(metalId)).to.equal(true);
+				});
+			});
```

Comment:
Please point out if I have misunderstood.

In `/metadata/metal/:metalId` the query is
metalId, mimeType, fileName, and download only.
In other words, I cannot control the text. (Same for seal).
Since text and seal should be given or not when decoding the metalID, I didn't think it was necessary to test here.
I think the test here is whether or not what we pass in the query makes the appropriate function call.

As for seal and text, I thought they were unnecessary here since they are done in `MetadataDb_spec.js`.

Reply:
I did it this way, but is it excessive?
I feel the first EXCEPT is sufficient.
```js
expect(mockServer.res.setHeader.calledWithExactly('Content-MetalText', 'db_text')).to.equal(true);
expect(mockServer.res.setHeader.calledWithExactly('Content-MetalText', 'db_fail_text')).to.equal(false);
```

Reply:
yes, first EXCEPT is sufficient